DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/streaming_data_interfaces/README.md

```diff
@@ -0,0 +1,272 @@
+   ```
+   mkdir build
+   cd build
+   cmake -G "NMake Makefiles" ..
+   ```
+   > **Note**: You can change the default target by using the command:
+   >  ```
+   >  cmake -G "NMake Makefiles" .. -DFPGA_DEVICE=<FPGA device family or FPGA part number>
+   >  ```
+   >
+   > This tutorial is only intended for use in the IP Authoring flow and does not support targeting an explicit FPGA board variant and BSP.
+
+3. Compile the design. (The provided targets match the recommended development flow.)
+
+   1. Compile and run for emulation (fast compile time, targets emulated FPGA device).
+      ```
+      nmake fpga_emu
+      ```
+   2. Generate the optimization report.
+      ```
+      nmake report
+      ```
+   3. Compile and run for simulation (fast compile time, targets simulated FPGA device).
+      ```
+      nmake fpga_sim
+      ```
+   4. Compile for FPGA hardware (longer compile time, targets an FPGA device).
+      ```
+      nmake fpga
+      ```
+> **Note**: If you encounter any issues with long paths when compiling under Windows*, you may have to create your ‘build’ directory in a shorter path, for example c:\samples\build.  You can then run cmake from that directory, and provide cmake with the full path to your sample directory.
+
+## Run the `Streaming Data Interfaces` Tutorial
+
+### On Linux
+
+1. Run the sample on the FPGA emulator (the kernel executes on the CPU).
+   ```
+   ./threshold_packets.fpga_emu
+   ```
+2. Run the sample on the FPGA simulator.
+   ```
+   CL_CONTEXT_MPSIM_DEVICE_INTELFPGA=1 ./threshold_packets.fpga_sim
+   ```
+	
+### On Windows
+
+1. Run the sample on the FPGA emulator (the kernel executes on the CPU).
+   ```
+   threshold_packets.fpga_emu.exe
+   ```
+2. Run the sample on the FPGA simulator.
+   ```
+   set CL_CONTEXT_MPSIM_DEVICE_INTELFPGA=1
+   threshold_packets.fpga_sim.exe
+   set CL_CONTEXT_MPSIM_DEVICE_INTELFPGA=
+   ```
+
+## Understanding the Tutorial
+
+In `threshold_packets.cpp`, two pipes are declared for implementing the streaming input and streaming output interfaces on a kernel which thresholds pixel values in an image. The streams use start of packet and end of packet signals to determine the beginning and end of the image.
+
+
+### Reading the Reports
+
+After compiling in the report flow, locate and open the `report.html` file in the `threshold_packets.report.prj/reports/` directory. Under the `Threshold` kernel in the System Viewer, the streaming in and streaming out interfaces can be seen, shown by the pipe read and pipe write nodes respectively. Clicking on either of these nodes gives further information for these interfaces in the Details pane. This pane will identify that the read is coming from `InPixel`, and that the write is going to `OutPixel`, as well as verifying that both interfaces have a width of 32 bits (corresponding to the `StreamingBeatT` type) and depth of 8 (which is the capacity that each pipe was declared with).
```

Comment:
The report says the interface has a width of 32 bits (= `sizeof(StreamingBeatT)`), but the documentation says _"Using the StreamingBeat struct with the uses_packets template parameter set to true adds two additional 1-bit signals to the Avalon interface, start_of_packet (sop), and end_of_packet (eop)."_ Would that not mean the interface should be 16 + 2 = 18 bits wide? or does it do some packing?

Reply:
Something to clarify with eng

Reply:
that sounds like a report defect to me.
the report should show the interface width as 16 bits wide (since you declared `StreamingBeatT` to have `ac_int<16, false>` as the underlying type).

If an 8-bit-wide data type results in the report showing a 24-bit interface, it sounds like those 1-bit signals are being parsed inside the compiler as 8-bit signals, and those are trimmed down after the report is generated.

Reply:
Okay. I will open a case on this